.github/workflows/README.md

```diff
@@ -0,0 +1,152 @@
+  the `Determine Build Args` job. This job fails if the building of 
+  the Docker images was unsuccessful. The Docker images built by this job
+  will not be published to GCR and are discarded as the job succeeds.
+
+
+3. `Publish`. This job is triggered when a PR is merged to a commit 
+  is pushed to the `master` branch. Similar to the `Test Images Build` job,
+  this job builds Docker images and fails if the build process was 
+  unsuccessful. However, in addition, this job pushes the built images
+  to GCR. To authorize access to GCR, this job assumes a GCP service 
+  account (SA) with read and write access to the GCR registry. The secrets 
+  related to the SA are defined as 
+  [encrypted environment secretes](docs.github.com/en/actions/security-guides/encrypted-secrets).
+
+    Note that ***under `broadinstitute/gatk-sv`, this job runs only 
+    after it is approved by one of the maintainers.***
+
+### Setup the `Deploy` Environment
+_This section describes configuring the `Deploy` environment to be used
+by the `Publish` job; it is intended for the use of the maintainers the
+`broadinstitute/gatk-sv`, and the admins of those forks of the repository 
+who would like to publish GATK-SV Docker images to their own registry
+via the CI/CD pipeline._
+
+An SA is used to authorize DIW to access GCR. (A future extension may 
+adopt an [OpenID Connect [OIDC]-based](https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/about-security-hardening-with-openid-connect)
+authentication and authorization.) In order to assume the SA, the `Publish`
+job needs the SA secrets (e.g., `private key` and `client email`) and 
+`project name`. This information is defined in a [GitHub environment](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment),
+and is exposed to the `Publish` job as encrypted environment secrets.
+The encrypted secrets are decrypted in the environment context and are 
+not exposed to the user code (if the norms of best practices are followed).
+GitHub's environment secrets are a subset of repository-wide secrets, 
+which is a subset of organization-level secrets. We encrypt SA credentials 
+as GitHub's environment secrets as they allow pausing the execution of any 
+action that accesses the environment until it is approved by assigned 
+individuals.
+
+In order to set up the `Deploy` environment, you may take the following steps:
+
+1. [Create an SA on GCP IAM](https://cloud.google.com/iam/docs/creating-managing-service-accounts#creating).
+   For simplicity, you may assign the service account the `Editor` role.
+   However, in order to follow the principles of the least privilege, 
+   you may assign the `Storage Object Admin`, `Storage Legacy Bucket Writer`,
+   and `Storage Object Viewer` as the minimum required permissions
+   ([ref](https://cloud.google.com/container-registry/docs/access-control)).
```

Comment:
A bit confused by this - it looks like you are suggesting the same examples from the linked doc, but for the purposes of deployment I think `Storage Object Creator` is necessary and sufficient for pushing images to GCR.

Reply:
Good point! I am not sure about the implementation details of the Docker push, and I am guessing `Storage Object Admin` is the minimum privilege to create buckets. In my experiments where the buckets exist and Docker push should only be triggering object creation, the following is the list of permissions used: 

```
storage.objects.create
storage.objects.delete
storage.objects.get
storage.objects.list
```

Referring to [this GCP documentation](https://cloud.google.com/storage/docs/access-control/iam-roles), it seems this list can be covered by a union of `Storage Object Creator` and `Storage Object Viewer` roles. We could experiment with that.

Reply:
I tested it using an SA with `Storage Object Creator` and `Storage Object Viewer` permissions, and Docker push fails with `403 Forbidden` error. So, I am guessing `Storage Object Admin` is required permission.